### ISSUE TITLE
[bug] Queue size lowering not redirecting the patients to other available idle rooms

### ISSUE THREAD

**Issue by divajsz** [CorsixTH/CorsixTH#2445]
### Describe the issue

I was testing 0.67 with the last but one map and I had 3 operation room built, all of them had 2-2-2 doctors in it. I set the queue size to 3 instead of 6 to distribute the patients among the 3 room. Emergency patients came, the queue filled up with 6 people. I tried to lower te max size but the patients were not moving to other room

Other interesting thing, when I had like 4 general practitioner rooms and I built a 5th one the patients came there where the new GP office estabilished and waited for the doctor to arrive.

### Steps to Reproduce

 I attached 4 savegame in chronological order for trial

[Saves.zip](https://github.com/CorsixTH/CorsixTH/files/13199513/Saves.zip)

### Expected Behaviour

I expect if I have idle rooms from the same type the and I lower the queue size e.g. 6 to 3 then the rest of the patients should go to an other idle room for treatment

### Save Game

Save game attached to report: Yes

### CorsixTH Version

0.67

### Operating System

windows 10

### Theme Hospital Version

Origin

### Gamelog.txt

```bash
Compiled with api_version: 2682, arch: x64, audio: true, os: windows
SDL renderer: direct3d
Lua 5.4 run with api version: 2682, game version: v0.66, savegame version: 170
Created game with savegame version 170.
Savegame version is 170 (v0.66), originally it was 170 (v0.66).

Kind of wierd... I just installed the 0.67 version on top
```

**Comment by lewri** [comment]
Funnily enough, I was convinced this was also broken in the original game too. Likely not enough weight going into patient redirection. Thanks for the report

**Comment by lewri** [comment]
https://github.com/CorsixTH/CorsixTH/blob/15368f125f16f36350d3bfb558374532eff1d9a1/CorsixTH/Lua/room.lua#L651-L665

Queue max size control is not very impactful maybe, could explain why.

What would be better is to start increasing the penalty proportionally to how much over size a queue is.

**Comment by ARGAMX** [comment]
Queue size lowering in vanilla TH does not redirect the patients to other available idle rooms too.

So by taking this into account this is more a **Feature request** than a **Bugfix**.

**Comment by lewri** [comment]
I agree. Though I also think Bullfrog likely messed this up in the original too. This would definitely be a QoL fix.

**Comment by ARGAMX** [comment]
I conducted an experiment, and it showed that when the queue size is reduced, patients are redistributed to other similar rooms.

Accordingly, I believe that what is stated in this issue is not valid. Either we fixed it, or there is some other reason. 

The only thing I can say is that this doesn’t happen instantly; it requires several game days.

You can check it yourself. Load this save in CorsixTH 0.70.1. There is an open queue for Operating Theatre with 6 people in it. Reduce the queue to 2–3 people and skip 5–10 in‑game days. You will see that people left the queue. 

 i.e. the mechanism has been triggered successfully.

[0.70.1_queue_of_6.sav.zip](https://github.com/user-attachments/files/32379643/0.70.1_queue_of_6.sav.zip)
